homeassistant/components/planifneige/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""
+Sensor for the City of Montreal's Planif-Neige snow removal APIs.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.planifneige/
+"""
+
+import logging
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.restore_state import RestoreEntity
+
+DEPENDENCIES = ['planifneige']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STREETID = 'streetid'
+CONF_STREETS = 'streets'
+CONF_DBPATH = 'database_path'
+
+ATTR_SENSOR = 'sensor'
+ATTR_STATES = 'states'
+
+DOMAIN = 'planifneige'
+
+STREET_STATE = {
+    '0': ['Snowed', 'mdi:snowflake'],  # street is snowed in
+    '1': ['Clear', 'mdi:road'],  # street is clear
+    '2': ['Planned', 'mdi:clock-outline'],  # street is planned for clearing
+    '3': ['Planned', 'mdi:clock-outline'],  # clearing planned with new date
+    '4': ['Snowed', 'mdi:snowflake'],  # to be planned, still snowed in
+    '5': ['Clearing', 'mdi:bulldozer'],  # trucks loading, still clearing
+    '10': ['Ploughed', 'mdi:road']  # street cleared, snow not loaded
+}
+
+PLANIFNEIGE_ATTRIBUTION = "Information provided by the City of Montreal "\
+    "(http://donnees.ville.montreal.qc.ca/dataset/deneigement)"
```

Comment:
Move this to the component and import it here. Also, we should remove the link and place it in the docs because it a pretty long attribution which will probably not look very nice in the frontend.